fossology/fossology-action
# chore(fossology action): Add fossology action

## Description

Set up a composite action to run fossology scanners in Github Actions.

### Changes
* Add a composite action for fossology scanners.
* Add README.md
* Add static folder with fossology logo.
* Add new inputs that can be passed by the user to provide customizability.

### How to test:
* Please look at `README.md` for detailed instructions.
* For PR's
```yaml
name: License scan on PR

on: [pull_request]

jobs:
  compliance_check:
    runs-on: ubuntu-latest
    name: Perform license scan
    steps:
    - name: Checkout
      uses: actions/checkout@v2
    
    - name: License check
      id: compliance
      uses: fossology/fossology-action@v1
      with:
        scan-mode: 'repo'
        scanners: 'nomos ojo'
        report_format: 'SPDX_JSON'
```

* For tags:
```yaml
name: License scan on tags

on: [tags]

jobs:
  compliance_check:
    runs-on: ubuntu-latest
    name: Perform license scan
    steps:
    - name: Checkout
      uses: actions/checkout@v2
    - name: License check
      id: compliance
      uses: fossology/fossology-action@v1
      with:
        scan-mode: 'differential'
        scanners: 'nomos ojo copyright keyword'
        from_tag: 'v003'
        to_tag: 'v004'
        report_format: 'SPDX_JSON'
```
CC @GMishx @shaheemazmalmmd @avinal  @Kaushl2208
